File: Modules/SEDO/EMS/SIE.Web.EMS.EquipMaint/Maintains/Plans/Commands/ImportMaintainPlanCommand.cs
using SIE.EMS.Maintains.Plans.ImportMaintainPlans;
using SIE.Web.Common.Import.Commands;
using SIE.Web.EMS.EquipMaint.Common.Commands;
using System;

namespace SIE.Web.EMS.EquipMaint.Maintains.Plans.Commands
{
    /// <summary>
    /// 保养计划导入命令
    /// </summary>
    public class ImportMaintainPlanCommand : ImportDataCommonCommand
    {
        /// <summary>
        /// 获取导入处理类型
        /// </summary>
        /// <returns></returns>
        protected override Type GetImportHandleType()
        {
            return typeof(ImportMaintainPlanHandle);
        }

        /// <summary>
        /// 设置模板文件信息
        /// </summary>
        /// <param name="importViewArgs"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override TemplatePathInfo SetTemplateFileName(ImportViewArgs importViewArgs, string scope)
        {
            const string templateFileName = "保养计划导入模板.xlsx";
            const string fileRelativePath = "Templates/" + templateFileName;
            return new TemplatePathInfo
            {
                FullPath = fileRelativePath,
                FileName = templateFileName
            };
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS.EquipMaint/Maintains/Plans/Commands/ImportMaintainPlanCommand.js
SIE.defineCommand('SIE.Web.EMS.EquipMaint.Maintains.Plans.Commands.ImportMaintainPlanCommand', {
    extend: 'SIE.Web.EMS.Common.Commands.ImportDataCommonCommand',
    meta: { text: "数据导入", group: "business", iconCls: "icon-Download icon-blue", model: "SIE.EMS.Maintains.Plans.ViewModels.MaintainPlanCheckDataViewModel" },

});